### PR TITLE
HW1 Task03

### PR DESCRIPTION
**Pull request by LinaKrevetochka**

### CODE REVIEW

File: Homework1/Task03/task03.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+We guarantee, that file exists and contains line-delimited integers.
+To read file line-by-line you can use this snippet:
+with open("some_file.txt") as fi:
+    for line in fi:
+        ...
+"""
+from typing import Tuple
+
+
+def find_maximum_and_minimum(file_name: str) -> Tuple[int, int]:
```
Comment (Cjkjvfnby) [inline Homework1/Task03/task03.py:14]: What is the memory complexity of this solution?   N is the number of rows in file.

File: Homework1/Task03/task03.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    nums = []
+    with open(file_name) as fi:
+        for line in fi:
```
Comment (Cjkjvfnby) [inline Homework1/Task03/task03.py:17]: you can use list comprehension here.  
```python
with open(file_name) as fi:
    nums = [ int(line) for line in f]
```

File: Homework1/Task03/task03.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            nums.append(int(line))
+    nums.sort()
+    return (nums[0], nums[len(nums) - 1])
```
Comment (Cjkjvfnby) [inline Homework1/Task03/task03.py]: Use index -1
```
return (nums[0], nums[-1])
```

You don't need parenthesis here

```
return nums[0], nums[-1]
```

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]

File: Homework1/Task03/task03.py (hunk 1 above)
Comment (LinaKrevetochka) [reply Homework1/Task03/task03.py:14]: The memory complexity is N. Do you mean that I shouldn't create a list and compare values in a loop?

Comment (Cjkjvfnby) [reply Homework1/Task03/task03.py:14]: yes, in this case this optimization looks reasonable.

File: Homework1/Task03/task03.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+with open("some_file.txt") as fi:
+    for line in fi:
+        ...
+"""
+from typing import Tuple
+
+
+def find_maximum_and_minimum(file_name: str) -> Tuple[int, int]:
+    with open(file_name) as fi:
+        min = max = int(fi.readline())
```
Comment (Cjkjvfnby) [inline Homework1/Task03/task03.py]: Please install `flake8-builtins` 

See detailed problem description ath the plugin page https://pypi.org/project/flake8-builtins/

**Review by Cjkjvfnby — CHANGES_REQUESTED** [review]
I am happy with the approach, please fix variable names.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**